addons/sale_loyalty/models/sale_order.py

```diff
@@ -582,6 +582,21 @@ def _write_vals_from_reward_vals(self, reward_vals, old_lines, delete=True):
         self.write({'order_line': command_list})
         return self.env['sale.order.line'] if delete else old_lines[len(reward_vals):]
 
+
+    def _is_better_global_discount_applied(self, current_reward, new_reward):
+        assert current_reward
+
+        def get_max_discount(reward):
+            if reward.discount_mode == 'per_order':
+                return reward.currency_id._convert(reward.discount, self.currency_id, self.company_id, fields.Date.today())
+            elif reward.discount_mode == 'percent':
+                discountable, _ = self._discountable_order(reward)
```

Comment:
I believe that the discountable amount of the sale order should also be taken into account for fixed rewards. 
You should return `min(discountable, fixed_reward)`